homeassistant/components/enocean/binary_sensor.py

```diff
@@ -111,3 +116,52 @@ def value_changed(self, packet):
                 "onoff": self.onoff,
             },
         )
+
+
+class EnOceanOccupancySensor(enocean.EnOceanDevice, BinarySensorDevice):
+    """Representation of EnOcean occupancy sensor.
+
+    Supported EEPs (EnOcean Equipment Profiles):
+    - A5-07-01 (Occupancy Sensor with Supply voltage monitor)
+    """
+
+    def __init__(self, dev_id, dev_name, device_class):
+        """Initialize the EnOcean binary sensor."""
+        super().__init__(dev_id, dev_name)
+        self._device_class = device_class
+        self._detection = False
+
+    @property
+    def name(self):
+        """Return the default name for the binary sensor."""
+        return self.dev_name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    def value_changed(self, packet):
+        """Fire an event with the data that have changed.
+
+        This method is called when there is an incoming packet associated
+        with this platform.
+
+        Example packet data:
+        - no occupancy detected       v-- pir status value, < 127 == off
+            ['0xa5', '0x7c', '0x0', '0x0', '0x9', '0x5', '0x3', '0x27', '0x7c', '0x0']
+        """
+
+        if packet.data[0] != 0xa5:
+            return
+
+        pir_status = packet.data[3]
+        self._detection = True if pir_status > 127 else False
+
+        self.hass.bus.fire(
```

Comment:
Entities should not fire events. Instead, this should be done inside the component part at `enocean/__init__.py`

Reply:
But then again, since you get signaled for both on and off, shouldn't it just be a binary sensor that sets the state to the latest detection event ?

Reply:
As I reasoned in the commit: "Only events are emitted, since the sensor doesn't have a state."
Problem ist, that these EnOcean sensors don't send a packet if the PIR falls below a certain threshold. I only witnessed packets whenever occupancy was detected, not when it "stopped".
Therefore I found a state change to be missleading and should be combined with additional logic on a higher level to derive actual occupancy status.

I also just followed the existing EnOcean sensor, which uses the same approach. I merely added another type.

If this is general HA policy this behavior should be changed for both I guess, which is not really the scope of this PR (or is it?)

Reply:
Yes, enocean should be changed to follow our current standards. We shouldn't extend the incorrect behavior.

If you can cleanly fire the event in the component it's ok to do that in this PR.

Otherwise we need a refactor first in a separate PR.